### PR TITLE
Add test function

### PR DESCRIPTION
**Pull request by VictorDuma-zz**

### CODE REVIEW

**Review by Arke64 — CHANGES_REQUESTED** [review]
Undo the changes to the newlines in the copyright header and at the end of the files. This PR looks like it is combining both an update to a graphics test and adding functionality to the BrainPad object itself. That should be two separate PRs. Additionally, for the graphics helpers, it looks like it just used to test, is a whole class in inc/src really needed? Can't we just put the font and draw char function in the test file itself?

**Comment by Arke64** [comment]
There are also conflicts that must be resolved

File: model/BrainPad.cpp [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*
 The MIT License (MIT)
 
-Copyright (c) 2016 Lancaster University, UK.
+Copyright (c) 2017 Lancaster University.
```
Comment (Arke64) [inline model/BrainPad.cpp]: Revert year

File: model/BrainPad.cpp [hunk 2]
```diff
@@ -16,13 +16,13 @@ all copies or substantial portions of the Software.
 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+
```
Comment (Arke64) [inline model/BrainPad.cpp]: Remove

File: model/BrainPad.h [hunk 3]
```diff
@@ -16,6 +16,7 @@ all copies or substantial portions of the Software.
 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+
```
Comment (Arke64) [inline model/BrainPad.h]: Remove

File: model/BrainPadIO.cpp [hunk 4]
```diff
@@ -16,6 +16,7 @@ all copies or substantial portions of the Software.
 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+
```
Comment (Arke64) [inline model/BrainPadIO.cpp]: Remove

File: model/BrainPadIO.cpp [hunk 5]
```diff
@@ -41,15 +42,20 @@ using namespace codal;
...
-    A1 (ID_PIN_P1, PA_1, PIN_CAPABILITY_AD),
-    A2 (ID_PIN_P2, PA_4, PIN_CAPABILITY_AD),
-    A3 (ID_PIN_P3, PB_0, PIN_CAPABILITY_AD),
-    A4 (ID_PIN_P4, PC_1, PIN_CAPABILITY_AD),
-    A5 (ID_PIN_P5, PC_0, PIN_CAPABILITY_AD),
-    led (ID_PIN_P6, PC_9, PIN_CAPABILITY_AD),
-    scl (ID_PIN_P9, PB_6, PIN_CAPABILITY_AD),
-    sda (ID_PIN_P10, PB_7, PIN_CAPABILITY_AD),
-    buttonA (DEVICE_ID_BUTTON_A, USER_BUTTON, PIN_CAPABILITY_DIGITAL)
++    scl (ID_PIN_SCL, PB_6, PIN_CAPABILITY_AD),
```
Comment (Arke64) [inline model/BrainPadIO.cpp]: Remove these extra '+' characters

File: model/BrainPadIO.h [hunk 6]
```diff
@@ -16,6 +16,7 @@ all copies or substantial portions of the Software.
 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+
```
Comment (Arke64) [inline model/BrainPadIO.h]: Remove

File: samples/main.cpp [hunk 7]
```diff
@@ -1,79 +1,223 @@
...
+void DrawText(int x, int y, char letter) {
+    int index = 5 * (letter - 32);
 
-    do {
-        if (dp < 0) {
-            dp = dp + 2 * (++x) + 3;
+    for (int h = 0; h < 5; h++) {
+        for (int v = 0; v < 8; v++) {
+            int show = (font[index + h] & (1 << v)) != 0;
+            DrawPoint(x + h  , y + v , show);
```
Comment (Arke64) [inline samples/main.cpp]: Remove extra space before commas

File: samples/main.cpp [hunk 8]
```diff
@@ -1,79 +1,223 @@
...
-    int dp = 1 - radius;
+void DrawText(int x, int y, char letter) {
+    int index = 5 * (letter - 32);
 
-    do {
-        if (dp < 0) {
-            dp = dp + 2 * (++x) + 3;
+    for (int h = 0; h < 5; h++) {
+        for (int v = 0; v < 8; v++) {
+            int show = (font[index + h] & (1 << v)) != 0;
```
Comment (Arke64) [inline samples/main.cpp]: Inline show

File: samples/main.cpp [hunk 9]
```diff
@@ -1,79 +1,223 @@
...
+void TestServo() {
+    brain.io.servoOne.setServoValue(180);
+    brain.sleep(200);
+    brain.io.servoOne.setServoValue(90);
+    brain.sleep(200);
+    brain.io.servoOne.setServoValue(0);
+    brain.sleep(200);
+}
+
+// Event test
```
Comment (Arke64) [inline samples/main.cpp]: Remove this commented out block

File: samples/main.cpp [hunk 10]
```diff
@@ -1,79 +1,223 @@
...
 
     while (true) {
-        brain.io.led.setDigitalValue(1);
-        brain.sleep(100);
-        brain.io.led.setDigitalValue(0);
-        brain.sleep(100);
+        TestButtons();
+        TestTemperatureSensor();
+        TestLightSensor();
+        // show the messages on the screen
```
Comment (Arke64) [inline samples/main.cpp]: Remove comment

**Review by Arke64 — CHANGES_REQUESTED** [review]

File: samples/main.cpp (hunk 8 above)
Comment (VictorDuma-zz) [reply samples/main.cpp]: Please, explain this

**Review by Arke64 — APPROVED** [review]

**Pull request merged**